analyses/long-format-table-utils/annotator/download-annotation-data.R

```diff
@@ -0,0 +1,205 @@
+      stop(paste0("mygene query returns non-character refseq.protein.\n",
+                  "Check query results. Revise download-annotation-data.R ",
+                  "to handle non-character values."))
+    }
+  })
+
+  # combimed vector of unique refseq.protein values
+  uniq_c_rm_null_rp_vec <- sort(unique(
+    purrr::reduce(rm_null_rp_vec_list, c, .init = character(0))))
+  # remove NA
+  rm_na_uniq_c_rm_null_rp_vec <- purrr::discard(
+    uniq_c_rm_null_rp_vec, is.na)
+  # keep only NP_### refseq.protein values
+  np_rp_vec <- purrr::keep(
+    rm_na_uniq_c_rm_null_rp_vec, function(x) stringr::str_detect(x, "^NP_"))
+
+  clp_np_rp_str <- paste(np_rp_vec, collapse = ",")
+
+  if (identical(clp_np_rp_str, "")) {
+    # no NP_### value for the gene query
+    # dplyr::summarise needs function return value to be character for a
+    # character column. as.character(NA) is still NA rather than "NA".
+    return(as.character(NA))
+  } else{
+    return(clp_np_rp_str)
+  }
+}
+# # test cases
+# collapse_rp_lists(list(NULL, NULL, NULL))
+# collapse_rp_lists(list())
+# collapse_rp_lists(list(c("NP_1", "NP_2"), c("NP_1", "NP_3")))
+# collapse_rp_lists(list(c("NX_1", "NA"), c("NP_1", "NP_3")))
+# collapse_rp_lists(list(c("NP_1", "NP_2"), NULL, c("NP_1", "NP_3")))
+# collapse_rp_lists(list(c("NP_3", "NP_2")))
+# collapse_rp_lists(list(c("NP_1", "NP_2"), NULL, c("NP_1", "NP_3", NA),
+#                        c(NA, NA, character(0)), c(NA, character(0)),
+#                        character(0)))
+
+
+
+# Collapse a name character vector from mygene.info query results
+#
+# Args:
+# - gn_vec: character vector of name values from mygene.info query results
+#
+# Returns a single character value of a comma-separated name value or NA
+collapse_name_vec <- function(gn_vec) {
+  # assert non-null list elements are all characters
+  if (!is.character(gn_vec)) {
+    stop(paste0("mygene query returns non-character refseq.protein.\n",
+                "Check query results. Revise download-annotation-data.R ",
+                "to handle non-character values."))
+  }
+  # remove NAs
+  uniq_rm_na_gn_vec <- sort(unique(purrr::discard(gn_vec, is.na)))
+
+  clp_gn_str <- paste(uniq_rm_na_gn_vec, collapse = ",")
+  if (identical(clp_gn_str, "")) {
+    # no non-NA name for the gene query
+    # dplyr::summarise needs function return value to be character for a
+    # character column. as.character(NA) is still NA rather than "NA".
+    return(as.character(NA))
+  } else {
+    return(clp_gn_str)
+  }
+}
+# # test cases
+# collapse_name_vec(c("a", "b"))
+# collapse_name_vec(c("a", "b", NA))
+# collapse_name_vec(c("a", "a", "a", "b", "d"))
+# collapse_name_vec(c("d", "a", "a", "b", "d"))
+# collapse_name_vec(c(NA, character(0)))
+# is.na(collapse_name_vec(c(NA, character(0))))
+# class(collapse_name_vec(c(NA, character(0))))
+# collapse_name_vec(c(NA, NA, NA, character(0)))
+# collapse_name_vec(character(0))
+# # folloing cases should fail
+# collapse_name_vec(c(1))
```

Comment:
I agree. It would be informative to have the expected return values.

The unit testing is now implemented in another framework, so the test cases can be removed. See comment <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/55#discussion_r674266765> for more details.